### PR TITLE
feat(cmd/influx): add --file option to query and task commands

### LINKED ISSUE #17356: Remove required '@' prefix for file paths in 'influx' CLI
__Proposal:__
Remove the required `@` prefix when specifying file paths in the `influx` CLI. It breaks any shell's native tab-completion.

__Desired behavior:__
I want to be able to use my shell's native tab-completion to quickly type file paths when using the `influx` CLI.

__Use case:__
Ease of use.

### PR DESCRIPTION
**Pull request by sranka**
Closes #17356

Describe your proposed changes here.

- `influx query`, `influx task create` and `influx task update` now support a new --file (-f) option
- flux is read from stdin when no argument or file is supplied
- backward compatibility: '@/path/to/flux' or '-' arguments are still supported

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by sranka** [comment]
@sanderson can you pls review/proofread help messages

File: cmd/influx/query.go [hunk 1]
```diff
@@ -2,30 +2,67 @@ package main
...
 
 func cmdQuery(f *globalFlags, opts genericCLIOpts) *cobra.Command {
-	cmd := opts.newCmd("query [query literal or @/path/to/query.flux]", fluxQueryF, true)
+	cmd := opts.newCmd("query [query literal or -f /path/to/query.flux]", fluxQueryF, true)
 	cmd.Short = "Execute a Flux query"
-	cmd.Long = `Execute a literal Flux query provided as a string,
-or execute a literal Flux query contained in a file by specifying the file prefixed with an @ sign.`
-	cmd.Args = cobra.ExactArgs(1)
+	cmd.Long = `Execute a literal Flux query provided as an argument or 
+in a file using the -f flag or via stdin`
```
Comment (sanderson) [inline cmd/influx/query.go:24]: Execute a Flux query provided via stdin or a file

File: cmd/influx/query.go [hunk 2]
```diff
@@ -2,30 +2,67 @@ package main
...
+	cmd.Args = cobra.MaximumNArgs(1)
 
 	queryFlags.org.register(cmd, true)
+	cmd.Flags().StringVarP(&queryFlags.file, "file", "f", "", "The path to the Flux query file")
```
Comment (sanderson) [inline cmd/influx/query.go:28]: ~The~ Path to Flux query file

File: cmd/influx/task.go [hunk 3]
```diff
@@ -45,14 +44,18 @@ var taskPrintFlags struct {
...
 }
 
 func taskCreateCmd(opt genericCLIOpts) *cobra.Command {
-	cmd := opt.newCmd("create [query literal or @/path/to/query.flux]", taskCreateF, true)
-	cmd.Args = cobra.ExactArgs(1)
+	cmd := opt.newCmd("create [script literal or -f /path/to/script.flux]", taskCreateF, true)
+	cmd.Args = cobra.MaximumNArgs(1)
 	cmd.Short = "Create task"
+	cmd.Long = `Create task with a Flux script provided as an argument or 
+in a file using the -f flag or via stdin`
```
Comment (sanderson) [inline cmd/influx/task.go:56]: Create a task with a Flux script provided via stdin or a file

File: cmd/influx/task.go [hunk 4]
```diff
@@ -45,14 +44,18 @@ var taskPrintFlags struct {
...
 
+	cmd.Flags().StringVarP(&taskCreateFlags.file, "file", "f", "", "The path to the Flux script file")
```
Comment (sanderson) [inline cmd/influx/task.go:58]: ~The~ Path to Flux script file

File: cmd/influx/task.go [hunk 5]
```diff
@@ -207,15 +210,20 @@ func taskFindF(cmd *cobra.Command, args []string) error {
...
 	status string
+	file   string
 }
 
 func taskUpdateCmd(opt genericCLIOpts) *cobra.Command {
 	cmd := opt.newCmd("update", taskUpdateF, true)
 	cmd.Short = "Update task"
+	cmd.Long = `Update task
+An optional Flux script can be provided as an argument or 
+in a file using the -f flag, '-' argument reads the Flux script from stdin.`
```
Comment (sanderson) [inline cmd/influx/task.go:221]: Update task status or script. Provide a Flux script via stdin or a file.

Is the "`-` from stdin" still true?

File: cmd/influx/task.go [hunk 6]
```diff
@@ -207,15 +210,20 @@ func taskFindF(cmd *cobra.Command, args []string) error {
...
 
 	registerPrintOptions(cmd, &taskPrintFlags.hideHeaders, &taskPrintFlags.json)
 	cmd.Flags().StringVarP(&taskUpdateFlags.id, "id", "i", "", "task ID (required)")
 	cmd.Flags().StringVarP(&taskUpdateFlags.status, "status", "", "", "update task status")
+	cmd.Flags().StringVarP(&taskUpdateFlags.file, "file", "f", "", "The path to the Flux script file")
```
Comment (sanderson) [inline cmd/influx/task.go:226]: ~The~ Path to Flux script file.

File: cmd/influx/task.go (hunk 5 above)
Comment (sranka) [reply cmd/influx/task.go:221]: Yes, both '-' and '@file' are still supported for backward compatibility reasons. Herein, If the user wants to update the script from std-in, "-" must be an argument to let the tool know that the flux script comes from stdin.

**Pull request merged**